openquakeplatform/openquakeplatform/static/js/irv_viewer_d3_tree.js

```diff
@@ -270,7 +284,61 @@
                     }
                 })
                 .attr("id", function(d) {return "operator-label-" + d.level;})
-                .attr("x", function(d) { return d.weight * CIRCLE_SCALE + 15; });
+                .attr("x", function(d) { return Math.abs(d.weight) * CIRCLE_SCALE + 15; })
+                .on("click", function(d) {
+                    onTreeElementClick(d);
+                });
+
+            // Render 'ignore weights' into a new line when present
+            nodeEnter.append("text")
+                .text(function(d) {
+                    if (d.children){
+                        var ignoreWeightsStr = '';
+                         if (d.operator.indexOf('ignore weights') != -1) {
+                            parts = d.operator.split('(');
+                            ignoreWeightsStr = '(' + parts[1];
+                        }
+                        return ignoreWeightsStr;
+                    }
+                })
+                .style("fill", function(d) {
+                    if (d.operator != undefined) {
+                        // Check for operators that ignore weights and style accordingly
+                        var color;
+                        if (d.operator.indexOf('ignore') != -1) {
```

Comment:
On my side I removed this check, because ignoreWeightStr is empty if the operator does not ignore weights. So, here I use color = '#660000' always.